src/Analysis/Ast/Impl/Types/PythonClassType.Generics.cs

```diff
@@ -285,6 +287,18 @@ internal partial class PythonClassType {
             }
         }
 
+        /// <summary>
+        /// Resolves bounds for generic type parameters
+        /// </summary>
+        private void ResolveBounds(PythonClassType classType) {
+            foreach(var key in classType._genericParameters.Keys.ToList()) {
```

Comment:
`var (key, parameterType) in classType._genericParameters.ToList()` then you won't need `classType._genericParameters[key]`. Also, `ResolveBounds` should belong to class since it accepts `classType` which is basically `this` and accesses private variables. So it really is `ResolveBounds()`

Reply:
I've removed the methods to handle return type resolution as we need it due to test failures.